esp-hal/src/i2s_parallel.rs

```diff
@@ -607,32 +614,23 @@ pub trait Instance: Signals + RegBlock {
     }
 }
 
-impl RegBlock for I2S0 {
-    fn register_block(&self) -> &'static RegisterBlock {
+impl Instance for I2S0 {
+    fn register_block(&self) -> &RegisterBlock {
         unsafe { &*I2S0::PTR.cast::<RegisterBlock>() }
     }
-}
-
-impl Signals for I2S0 {
-    fn get_peripheral(&self) -> crate::system::Peripheral {
-        crate::system::Peripheral::I2s0
-    }
-
-    fn get_dma_peripheral(&self) -> crate::dma::DmaPeripheral {
-        crate::dma::DmaPeripheral::I2s0
-    }
 
     fn ws_signal(&self) -> OutputSignal {
         OutputSignal::I2S0O_WS
     }
     fn data_out_signal(&self, i: usize, bits: u8) -> OutputSignal {
+        assert!(
+            bits == 8 || bits == 16,
+            "Number of bits must be 8 or 16, got {}",
+            bits
+        );
+
         // signals for 8bit and 16bit both start at an offset of 8 for I2S0
-        let offset = match bits {
-            8 => 8,
-            16 => 8,
-            _ => panic!("Invalid number of bits"),
-        };
-        match i + offset {
+        match i + 8 {
```

Comment:
@liebman what was the logic here, are the bottom 8 signals not connected to the peripheral? Where is this documented, where did your information come from?

Reply:
The bottom 8 signals are only used if the with is 24bits.  This is in the esp-idf implementation as well and the TRM does not document this :-/.

Reply:
Okay I found [this](https://github.com/espressif/esp-idf/blob/9106c43accd9f5e75379f62f12597677213f5023/components/esp_lcd/i80/esp_lcd_panel_io_i2s.c#L701) which explains the +8, but what about I2S1? Is esp-idf buggy?

Reply:
yea - I2S1 should be the same (somehow I lost that)